### ISSUE TITLE
highlight bad test functions

### ISSUE THREAD

**Issue by happygiraffe** [golang/lint#161]
I've just seen some test code which defined:

```
func BlahTest(t *testing.T) { … }
```

This won't work unless the function name begins with `Test`.  I propose a warning if:

- A function should trigger warning when:
    1. In a `_test.go` file
    2. Exported
    3. Takes a single `testing.T` arg
    4. Name does not start with `Test`

**Comment by dsymonds** [comment]
There was a recently added `vet` check that does this. It belongs there because this is more of a correctness check than a style check.
